src/traces/parcoords/parcoords.js

```diff
@@ -0,0 +1,762 @@
+        .append('canvas')
+        .attr('class', function(d) {return 'parcoords-lines ' + (d.context ? 'context' : d.pick ? 'pick' : 'focus');})
+        .style('box-sizing', 'content-box')
+        .style('transform', 'translate(' + (-c.overdrag) + 'px, 0)')
+        .style('float', 'left')
+        .style('clear', 'both')
+        .style('left', 0)
+        .style('overflow', 'visible')
+        .style('position', function(d, i) {return i > 0 ? 'absolute' : 'absolute';})
+        .filter(function(d) {return d.pick;})
+        .on('mousemove', function(d) {
+            if(linePickActive && d.lineLayer && callbacks && callbacks.hover) {
+                var event = d3.event;
+                var cw = this.width;
+                var ch = this.height;
+                var translateY = d.model.key * (d.model.height + d.model.pad.t + d.model.pad.b) + d.model.translateY;
+                var x = event.layerX - d.model.pad.l + c.overdrag - d.model.translateX;
+                var y = event.layerY - d.model.pad.t - translateY;
+                if(x < 0 || y < 0 || x >= cw || y >= ch) {
+                    return;
+                }
+                var pixel = d.lineLayer.readPixel(x, ch - 1 - y);
+                var found = pixel[3] !== 0;
+                // inverse of the calcPickColor in `lines.js`; detailed comment there
+                var curveNumber = found ? pixel[2] + 256 * (pixel[1] + 256 * pixel[0]) : null;
+                var eventData = {
+                    x: x,
+                    y: y,
+                    clientX: event.clientX,
+                    clientY: event.clientY,
+                    dataIndex: d.model.key,
+                    curveNumber: curveNumber
+                };
+                if(curveNumber !== lastHovered) { // don't unnecessarily repeat the same hit (or miss)
+                    if(found) {
+                        callbacks.hover(eventData);
+                    } else if(callbacks.unhover) {
+                        callbacks.unhover(eventData);
+                    }
+                    lastHovered = curveNumber;
+                }
+            }
+        });
+
+    parcoordsLineLayer
+        .style('padding', function(d) {
+            var p = d.viewModel.model.pad;
+            return p.t + 'px ' + p.r + 'px ' + p.b + 'px ' + p.l + 'px';
+        })
+        .attr('width', function(d) {return d.viewModel.model.canvasWidth;})
+        .attr('height', function(d) {return d.viewModel.model.canvasHeight;})
+        .style('width', function(d) {return (d.viewModel.model.width + 2 * c.overdrag) + 'px';})
+        .style('height', function(d) {return d.viewModel.model.height + 'px';})
+        .style('opacity', function(d) {return d.pick ? 0.01 : 1;});
+
+    svg.style('background', 'rgba(255, 255, 255, 0)');
+    var parcoordsControlOverlay = svg.selectAll('.parcoords')
+        .data(vm, keyFun);
+
+    parcoordsControlOverlay.exit().remove();
+
+    parcoordsControlOverlay.enter()
```

Comment:
While trying to plot multiple `parcoords` traces on one graph, this here throws an error for

```js
Tabs.getMock('gl2d_parcoords_large', (err, mock) => {
  var data = [
    Lib.extendDeep({}, mock.data[0]),
    Lib.extendDeep({}, mock.data[0])
  ]

  data[0].domain = { y: [0, 0.5] }
  data[1].domain = { y: [0.5, 1] }

  Plotly.plot(Tabs.fresh(), data, mock.layout)
})
```